tap_shiphero/client.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime, timedelta
+
+import backoff
+import requests
+from singer import metrics
+from ratelimit import limits, RateLimitException, sleep_and_retry
+
+class Server5xxError(Exception):
+    pass
+
+class ShipHeroClient(object):
+    BASE_URL = 'https://api-gateway.shiphero.com/v1.2/general-api'
+
+    def __init__(self, token, user_agent):
+        self.__token = token
+        self.__user_agent = user_agent
+        self.__session = requests.Session()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.__session.close()
+
+    @backoff.on_exception(backoff.expo,
+                          [Server5xxError,
+                           RateLimitException],
+                          max_tries=5,
+                          factor=2)
+    @sleep_and_retry
+    @limits(calls=2, period=1)
```

Comment:
Do we know if ShipHero responds with a 429 when you get ratelimited?

Reply:
@KAllan357 I've only ever been able to get it to 5xx when I make more than two calls concurrently, but could try more testing. It's a incredibly slow API, with queries of more than a daily interval timing out. I got it down to an average of 9-10 seconds per call